compiler/rustc_type_ir/src/sty.rs

```diff
@@ -0,0 +1,471 @@
+use crate::DebruijnIndex;
+use crate::FloatTy;
+use crate::IntTy;
+use crate::Interner;
+use crate::TyDecoder;
+use crate::TyEncoder;
+use crate::UintTy;
+
+use rustc_serialize::{Decodable, Encodable};
+
+/// Defines the kinds of types used by the type system.
+///
+/// Types written by the user start out as [hir::TyKind](rustc_hir::TyKind) and get
+/// converted to this representation using `AstConv::ast_ty_to_ty`.
+#[allow(rustc::usage_of_ty_tykind)]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+//#[derive(TyEncodable, TyDecodable)]
+//#[derive(HashStable)]
```

Comment:
Can you remove these and instead leave a comment on where to find these impls and why?